### PR TITLE
added test for SubjectGroup

### PR DESCRIPTION
**Pull request by harshkothari410**
- getSubjectsToShow
- reset
- trackSampleAspectAndSubject

### CODE REVIEW

**Comment by iamigo** [comment]
lots of code style issues - can you fix and then I'll review? I don't want to spend a ton of time just commenting on spacing and stuff

**Comment by harshkothari410** [comment]
@iamigo fixed now. Sorry jscs was some how disabled thats why I missed it.

File: MultiTable/test/SubjectGroup.js [hunk 1]
```diff
@@ -142,4 +142,275 @@ describe('SubjectGroups Tests', () => {
...
+
+      subjectGroup.trackSampleAspectAndSubject(sample1);
+      expect(subjectGroup.subjectsToShow.size).to.equal(1);
+      expect(subjectGroup.subjectsToShow.has('abc')).to.be.true;
+      expect(subjectGroup.aspectsToShow.size).to.equal(1);
+      expect(subjectGroup.aspectsToShow.has('test1')).to.be.true;
+      return done();
+    });
+
+    it('showAll is off and OK status present then do not show that' +
```
Comment (jgraff2) [inline MultiTable/test/SubjectGroup.js]: this is the same as on line 221, right?

File: MultiTable/test/SubjectGroup.js [hunk 2]
```diff
@@ -142,4 +142,275 @@ describe('SubjectGroups Tests', () => {
...
+      subjectGroup.trackSampleAspectAndSubject(sample1);
+      expect(subjectGroup.subjectsToShow.size).to.equal(0);
+      expect(subjectGroup.subjectsToShow.has('abc')).to.be.false;
+      expect(subjectGroup.aspectsToShow.size).to.equal(0);
+      expect(subjectGroup.aspectsToShow.has('test1')).to.be.false;
+      return done();
+    });
+
+    it('Subject not present', (done) => {
+      subjectGroup.subjectsToShow = new Set([]);
```
Comment (jgraff2) [inline MultiTable/test/SubjectGroup.js]: You're already doing this in the beforeEach.

File: MultiTable/test/SubjectGroup.js [hunk 3]
```diff
@@ -142,4 +142,275 @@ describe('SubjectGroups Tests', () => {
...
+      subjectGroup.trackSampleAspectAndSubject(sample1);
+      expect(subjectGroup.subjectsToShow.size).to.equal(0);
+      expect(subjectGroup.subjectsToShow.has('abc')).to.be.false;
+      expect(subjectGroup.aspectsToShow.size).to.equal(1);
+      expect(subjectGroup.aspectsToShow.has('test1')).to.be.true;
+      return done();
+    });
+
+    it('Aspect not present', (done) => {
+      subjectGroup.subjectsToShow = new Set([]);
```
Comment (jgraff2) [inline MultiTable/test/SubjectGroup.js]: same.

File: MultiTable/test/SubjectGroup.js [hunk 4]
```diff
@@ -142,4 +142,275 @@ describe('SubjectGroups Tests', () => {
...
+      subjectGroup.trackSampleAspectAndSubject(sample1);
+      expect(subjectGroup.subjectsToShow.size).to.equal(1);
+      expect(subjectGroup.subjectsToShow.has('abc')).to.be.true;
+      expect(subjectGroup.aspectsToShow.size).to.equal(0);
+      expect(subjectGroup.aspectsToShow.has('test1')).to.be.false;
+      return done();
+    });
+
+    it('Aspect and Subject not present', (done) => {
+      subjectGroup.subjectsToShow = new Set([]);
```
Comment (jgraff2) [inline MultiTable/test/SubjectGroup.js]: same.

File: MultiTable/test/SubjectGroup.js (hunk 1 above)
Comment (jgraff2) [reply MultiTable/test/SubjectGroup.js]: @harshkothari410 can you remove this whole test too? Unless I'm missing something, it's a duplicate of the test on line 221.

**Pull request merged**